Use original Error object for uncaught errors when provided

mgiuffrida opened the pull request:
Include stack trace of async errors by not throwing away the Error passed by global.onerror.

Fixes #958, the same issue referenced at https://github.com/mochajs/mocha/issues/815#issuecomment-152226789.

mgiuffrida commented:
@freshp86 this helps with stack traces in Chrome's WebUI tests

boneskull commented:
@mgiuffrida `mocha.js` is generated by `make`; you'll need to apply this change in `lib/` somewhere (or maybe `browser-entry.js`)

going to try to release v3.0.0 tonight, so you might want to wait until that happens before retrying.

mgiuffrida commented:
@boneskull thanks, that was silly of me.

From the actual `mocha.js` log it seems I shouldn't try to commit a built version; i've moved the change to `browse-entry.js` instead and rebased.

boneskull commented on browser-entry.js:
```diff
@@ -51,8 +51,11 @@ process.removeListener = function(e, fn){
   if ('uncaughtException' == e) {
-    global.onerror = function(err, url, line){
-      fn(new Error(err + ' (' + url + ':' + line + ')'));
+    global.onerror = function(err, url, line, col, errObj){
```
should `col` be leveraged?

mgiuffrida replied:
It isn't currently, and it normally isn't in JS, right?

boneskull commented:
Closing for inactivity; please reopen if continuing.  Thanks!
